### DISCUSSION TITLE
TableRelation should be removed in Tables with ObsoleteState Removed

### DISCUSSION THREAD

**Discussion by pri-kise** [StefanMaron/BusinessCentral.LinterCop#655] [New Rule]
I suggest a warning when there are TableRelations in obsoleted Tables.

In the example below:
When you try to rename the Company then you receive an error
```al
table 50000 "PTE Company Mapping"
{
    Caption = 'Company Mapping';
    ObsoleteReason = 'Replace by table "PTE Company Name Mapping"';
    ObsoleteState = Removed;

    fields
    {
        field(1; "Company Name"; Text[30])
        {
            Caption = 'Company Name';
            DataClassification = CustomerContent;
            NotBlank = true;
            TableRelation = Company.Name; // Error when Company is renamed
        }
        field(2; "External Company Name"; Text[250])
        {
            Caption = 'External Company Name';
            DataClassification = CustomerContent;
        }
    }

    keys
    {
        key(PK; "Company Name")
        {
            Clustered = true;
        }
    }
}
```

**Comment by rvanbekkum** [comment]
What is the error that you'll get?

**Reply by pri-kise** [reply]
> Table "PTE Company Mapping" is obsoleted, reason: Replace by table "PTE Company Name Mapping"
> The Page "Edit - Companies" has to close.

My colleague reported this error for our AL solution for BC14.
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/56249171/a830f075-8c47-49f5-aa7e-86595c442cdf)

**Comment by NKarolak** [comment]
If I knew this place to submit new suggestion, I might have suggested this one myself as well :-)

Additionally, removed fields shouldn't have any triggers either.
